src/routes/users.rs

```diff
@@ -121,6 +122,43 @@ pub async fn mods_list(
     }
 }
 
+#[get("teams")]
+pub async fn teams(
+    req: HttpRequest,
+    info: web::Path<(UserId,)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let id: crate::database::models::UserId = info.into_inner().0.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool).await.ok();
+
+    let results;
+
+    if let Some(user) = current_user {
+        if user.id.0 == id.0 as u64 {
+            results =
+                crate::database::models::TeamMember::get_from_user_private(id, &**pool).await?;
+        } else {
+            results =
+                crate::database::models::TeamMember::get_from_user_public(id, &**pool).await?;
+        }
+    } else {
+        results = crate::database::models::TeamMember::get_from_user_public(id, &**pool).await?;
+    }
+
+    let team_members: Vec<crate::models::teams::TeamMember> = results
+        .into_iter()
+        .map(|data| crate::models::teams::TeamMember {
+            user_id: data.user_id.into(),
+            name: data.name,
+            role: data.role,
+            permissions: Permissions::from_bits_truncate(data.permissions as u64),
```

Comment:
This shouldn't return permissions unless the user is the current logged in user.